Migrate intelv2 aa1 to implement 2

pinterid opened the pull request:
**Checklist:**

 <!--- Make sure your PR is documented and tested before submission. Put an `x` in all the boxes that apply: -->
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?


**Type of Changes:**

 <!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] New feature (non-breaking change which adds functionality)
 - [ ] Documentation (non-breaking change which adds documentation)
 - [x] Breaking change (fix or feature that would cause existing functionality to change)


**What is the current behavior? (link to any open issues here)**
- Currently the front does not work with the latest intel. #124 
- Currently there is no company page. #98 
- Currently there is no media engineer page. #96 
- Currently it is not possible to follow other users. #95 
- Currently it is not possible to like other profiles. #94 
- Currently it is not possible to redeem achievements- #93 


**What is the new behavior (if this is a feature change)?**
- Now the front works with the latest intel. #124 
- Now there is a company page. #98 
- Now there is a media engineer page. #96 
- Now it is possible to follow other users. #95 
- Now it is possible to like other profiles. #94 
- Now it is possible to redeem achievements- #93 


**Other information**:
- Ref: https://github.com/snek-at/intel/releases/tag/0.1.2

Aichnerc committed:
Add new intel and client

Yeet old out of existence, yeet new into existence.

Aichnerc committed:
Add new register

The register component has been reworked.

Aichnerc committed:
Improve register

The labels have been improved.

schettn committed:
Remove all actions and implement user actions

All actions have been removed.
User actions like login, logout, and getPerson
have been implemented.

schettn committed:
Adjust components to work with new user actions

LoginForm and HomePage have been adjusted to
work with the new user actions and state definition.

schettn committed:
Merge remote-tracking branch 'origin/migrate-intelv2-aa1' into migrate-intelv2-aa1

schettn committed:
Comment out a bit

Code was commented out to avoid errors caused by
deleting the actions.

schettn committed:
Add person get action

Now it is possible to fetch persons which get stored in the store
as fetchedPerson.

schettn committed:
Rename import path for logout action

The logout action has been imported from another
location.

schettn committed:
Update profile page

Cleanup and person action integration

schettn committed:
Fix wrong dispatches in person action

Incorrect dispatches in the person action have
been fixed.

Aichnerc committed:
Fix modals

Some non existent action has been removed.

Aichnerc committed:
Add ConnectModal

The modal to connect other accounts has been added.

Aichnerc committed:
Improve ConnectModal

Some functions have been introduced.

schettn committed:
Update search functionallity

Now it is possible again to search users.

schettn committed:
Add general reducers

A action for getting all gitlab servers for registration
has been added.

Aichnerc committed:
Improve ConnectModal

Some more functionality has been added.

schettn committed:
Improve store

Types have been improved.
Naming has been improved.

schettn committed:
Improve general actions

The get gitlab servers action is now exported correctly.
Also undefined variables have been removed.

schettn committed:
Fix naming issue

The reducers are now correctly added to the root
reducer.

Aichnerc committed:
Unyeet registration

The registration has been unyeeted.

schettn committed:
Add registration action

The registration action has been added.
It calls the login action after.

schettn committed:
Move person actions to general action

Get persons brief and get person have been moved
to the general actions.
The person actions should only contain actions for the
currently logged person.

schettn committed:
Merge remote-tracking branch 'origin/migrate-intelv2-aa1' into migrate-intelv2-aa1

schettn committed:
Add some more actions and reducers

Following actions and their reducers have been added:
- All Achievements
- Person Settings
- Person Meta Link
- Person Profiles
- Person Add Profile
- Person Delete Profile
- Person Update Profile
- Person Instagram Posts
- Person Process Profiles
- Person Follow
- Person Unfollow
- Person Like
- Person Unlike

schettn committed:
Add enterprise actions and reducers

Following actions and their reducers have been added:
- Enterprise General
- Enterprise Projects
- Enterprise Users

schettn committed:
Add achievement redeem action and reducer

Following actions and their reducers have been added:
- Redeem Achievement

schettn committed:
Fix typo in store types

A type has been fixed. Yey.

schettn committed:
Add missing register export

The export of register in the user action
has been added.

schettn committed:
Add valid username check

Now it is possible to check if a specific
username is already taken.

schettn committed:
Add talk actions and reducers

Following actions and their reducers have been added:
- Get Talk
- Get Talks
- Add Talk
- Delete Talk
- Update Talk
- Add Talk Comment
- Delete Talk Comment
- Update Talk Comment

schettn committed:
Adjust component name, location and adjust for new data structure

Now it is possible to view the person page with the new data
structure.

schettn committed:
Improve talks and some more improvements

- The add and delete talk action have been moved form
talks to person.
- GetPerson has been moved from general to person.
- Some improvements in displaying the correct data
has been done.

schettn committed:
Improve setting page

The setting handle has been improved.
Also the settings now work with intelv2.

schettn committed:
Fix navbar avatar image

Now there is the correct avatar image in the navbar.

schettn committed:
Add optional changing for logged user

Now there are more check for the logged user.

schettn committed:
Fix logout navbar issue

Now the navbar behaves properly when logging out.

schettn committed:
Improve talks tab


Due to the immutable of talkList the list had to be
copied.
Ref: https://github.com/apollographql/apollo-client/pull/5153

schettn committed:
Optimise settings handle change

The handle change method has been improved.
It can handle now also not only events.

schettn committed:
Add support for uploading a avatar image

Now it is possible to upload a avatar image in the settings.

schettn committed:
Fix talk tab iframe rendering error

Now it is possible to list the talk tab entries.

schettn committed:
Improve setting change check

Now it is possible to reactivate the save button.

schettn committed:
Set loggedusers person after updating settings

Now the logged user is updated too when the settings
are updated.

schettn committed:
Merge branch 'update-component-structure-aa1' into migrate-intelv2-aa1

schettn committed:
Merge branch 'implement-2' into migrate-intelv2-aa1

schettn committed:
Improve company page data

The action and reducers related to company page
has been improved.

schettn committed:
Optimise registration and login

Now it is possible to register a new user.
Hashing has been moved to the login and register action.

schettn committed:
Improve loading of person page

Now the person page shows a correct loading component
instead of the previous profile.

schettn committed:
Change achievements to achievement

The notation has been fixed.

schettn committed:
Add profiles to settings

Now it is possible to list all profiles on the settings
page.

schettn committed:
Update project and projects tab

Now projects can be listed in the projects tab without crashing.

schettn committed:
Add profile processing in person page

When a logged user visits his person page,
profile processing is started.

schettn committed:
Remove legacy code and console logs

Unused code and console logs have been removed.

schettn committed:
Fix project owner name

Now the correct project owner name is displayed.

schettn committed:
Fix info card languages

Now the person languages gets displayed in the info card.

schettn committed:
Integrate overview tab and calendar 3d

Now it is possible to view the 3d calendar again.

schettn committed:
Integrate calendar2d, contrib radar and latest activity chart

The components work again, but they are still a bit sloppy 
implemented.

pinterid committed:
Add to continue style

The style sheet for the to continue modal was added.

pinterid committed:
Add to continue modal

The to continue component was added.

pinterid committed:
Update exports

The to continue modal is now being exported.

pinterid committed:
Add follow styles

The style sheet for the follow modal was added.

pinterid committed:
Add follow modal

The follow modal component was added.

pinterid committed:
Update exports

The follow modal is now being exported.

pinterid committed:
Import modals

The 2 new modals are now being imported.

pinterid committed:
Import actions

The follow and unfollow actions are now being imported.

pinterid committed:
Update redux mapping

The redux mapping was updated to the needs.

pinterid committed:
Update imports

The MDB component imports were updated.

pinterid committed:
Update state

The state for the component was updated.

pinterid committed:
Add follow function

The function for following persons was added.

pinterid committed:
Add unfollow function

The function for unfollowing a person was added.

pinterid committed:
Add follow check function

A function was added for checking the follow status.

pinterid committed:
Add component function

Component function were added to keep the state up to date.

pinterid committed:
Add modal handle function

The function for handling the modals was added.

pinterid committed:
Implement modals

The modals are now implemented.

pinterid committed:
Implement follow

It is now possible to follow other users.

pinterid committed:
Fix settings bug

A settings bugs was fixed.

schettn committed:
Integrate movable pool

Now it is possible to reorganise your components.

pinterid committed:
Merge branch 'add-follow-aa1' into migrate-intelv2-aa1

pinterid committed:
Merge remote-tracking branch 'origin/migrate-intelv2-aa1' into migrate-intelv2-aa1

schettn committed:
Only render charts if statistics data is present

Now charts are only rendered if there is statistic data.

schettn committed:
Hide diagram labels if diagrams are not visible

The charts are not displayed if the required data
is not available.

schettn committed:
Add more settings

More settings have been added to the settings page and
were also applied to the overview tab.
The settings page has been improved by moving the display
settings to "customization" tab.
Display settings related to github and gitlab are disabled
when there is no github or gitlab profile present.

schettn committed:
First attempt to fix company page

First data of a enterprise is displayed now.

pinterid committed:
Fix person actions

The like and unlike actions were fixed.

pinterid committed:
Update follow modal

The follow modal was updated to deny unauthenticated actions.

pinterid committed:
Add likes modal styles

The style sheet for the likes modal was added.

pinterid committed:
Add likes modal

The likes modal component was added.

pinterid committed:
Update exports

The likes modal is now being exported.

pinterid committed:
Update imports

The info card imports were updated.

pinterid committed:
Implement likes modal

The likes modal was implemented.

pinterid committed:
Update redux mapping

The redux mapping was updated to support like and unlike.

pinterid committed:
Add like function

The function for liking persons was added.

pinterid committed:
Add unlike function

The function for unliking persons was added.

pinterid committed:
Add like checker

The function for checking if a person has likes was added.

pinterid committed:
Implement like component

The like component was implemented.

pinterid committed:
Merge branch 'add-like-aa1' into migrate-intelv2-aa1

pinterid committed:
Merge remote-tracking branch 'origin/migrate-intelv2-aa1' into migrate-intelv2-aa1

pinterid committed:
Update sign out

The sign out now no longer redirects.

pinterid committed:
Fix link ussues

The to param was added.

Aichnerc committed:
Improve settings menu

The menu items now appear clickable.

Aichnerc committed:
Improve settings profile picture selector

The mouse now switches to a pointer when hovering the image.

schettn committed:
Merge remote-tracking branch 'origin/migrate-intelv2-aa1' into migrate-intelv2-aa1

schettn committed:
Merge remote-tracking branch 'origin/migrate-intelv2-aa1' into migrate-intelv2-aa1

schettn committed:
Fix invalid login issue

Now it is possible to unsuccessfully login.

Aichnerc committed:
Improve company page

The enterprise overview tab has been adjusted and no longer brakes.

Aichnerc committed:
Improve enterprise project modal

The project modal has been improved. Projects and users are now being queried.

Aichnerc committed:
Fix fetching users

Instead of users, projects were loaded and returned.

Aichnerc committed:
Improve user modal

Some data has been adjusted. The code statistics have been improved.

Aichnerc committed:
Add code statistics to enterprise overview

The code statistics have been added to the enterprise overview page.

Aichnerc committed:
Add imprint

The imprint has been reworked and the details are now being shown.

Aichnerc committed:
Improve overview page

The tabs have been renamed and restructured.

Aichnerc committed:
Improve company page

Some more info is now being shown.

schettn committed:
Add fake data for enterprise users

Now  a fake username and password is displayed if
the user is a anonymous one.

schettn committed:
Add dynamic avatar image to company page

Now the company page no longer contains a
static image. Instead the company avatar image is
displayed.

schettn committed:
Improve instagram posts action

Now the instagram post action returns 
the posts.

Aichnerc committed:
Fix projects

The project language chart has been fully fixed.

schettn committed:
Remove duplicate status on person page

Now the status is only displayed once.

schettn committed:
Improve calendars

Now all calendar years are displayed properly.

schettn committed:
Reverse the reversing

Now the years list is not reversed anymore.
Old: [2020, 2019, 2018 ] Current
New: [2018, 2019, 2020 ] Current

Aichnerc committed:
Implement connections

It is now possible to add GitHub and Instagram. GitLab is still WIP.

Aichnerc committed:
Add connect modal styling

Styling has been added to the connect modal.

Aichnerc committed:
Implement gitlab

Gitlabs can now be added.

Aichnerc committed:
Improve gitlab modal

The button is now disabled unless the required info is given.

schettn committed:
Improve profile processing

A 30 second lock has been implemented which
occurs after profile processing is successful.
If you are on your own profile at the time the
lockdown is released, it will be re-fetched
immediately. Otherwise when entering the profile.

Aichnerc committed:
Implement media sections functionality

Some more functionality has been introduced like a young kitten to a new flat.

Aichnerc committed:
Implement video and song gallery

The two galleries have been added and are now working.

Aichnerc committed:
Improve instagram selection

The selector now works with real data.

Aichnerc committed:
Hide media section buttons for diff origin users

The buttons are now only visible on the own user profile page.

Aichnerc committed:
Rework editing system

The buttons have been altered.

Aichnerc committed:
Improve Instagram selection

The selection should now be saved.

Aichnerc committed:
Improve image selector

Meh.

Aichnerc committed:
Improve YouTube adding

The URL is now being used instead of video id.

Aichnerc committed:
Improve upload modal

The modal has been visually improved.

Aichnerc committed:
Improve upload modal

Clickable has been added.

Aichnerc committed:
Implement video removal

Videos can now be removed.

schettn committed:
Improve upload modal

Now there is a independent upload modal which it is not limited to talk uploading.
Imgur and anonfiles are available as storage engines.

Aichnerc committed:
Improve video removal

The id is now being passed on on creation.

schettn committed:
Merge remote-tracking branch 'origin/migrate-intelv2-aa1' into migrate-intelv2-aa1

schettn committed:
Merge remote-tracking branch 'origin/migrate-intelv2-aa1' into migrate-intelv2-aa1

Aichnerc committed:
Implement song removal

Songs can now be yeeted.

schettn committed:
Remove unused import form upload modal

A unused import has been removed.

schettn committed:
Improve import of talks tab

The import region has been improved.

schettn committed:
Add custom image upload

Now it is possible to upload custom images to imgur.

Aichnerc committed:
Improve image galleries

The galleries have been visually improved.

Aichnerc committed:
Implement image upload

It is now possible to upload images to IMGUR.

Aichnerc committed:
Implement picture removal

It is now possible to remove pictures.

Aichnerc committed:
Implement connect modal

The modal is being shown if no profiles are present.

Aichnerc committed:
Improve connect modal

The local storage entry is now being set on toggle.

schettn committed:
Add basic load more instagram posts functionality

Now it is possible to load more instagram posts. This feature
is now fully supported by intel, therefore the load more button,
is disabled now.

schettn committed:
Fix state setting on load more

Now the state is set correctly.

Aichnerc committed:
Fix connect modal

The modal now only shows up if you have no present profiles.

Aichnerc committed:
Improve code quality

Some console logs have been removed. The Instagram selection modal has been improved to make it more stable.

schettn committed:
Improve remove custom image function

Now when a image is deleted, it will also be deleted from imgur.

pinterid committed:
Add achievement component

The achievement component was added.

pinterid committed:
Update exports 

The achievement component is now being exported.

pinterid committed:
Improve code quality

An unnecessary console.log was removed.

pinterid committed:
Add achievement tabs styles

The stylesheet for the achievement tab was added.

pinterid committed:
Add achievement tab

The achievement tab was added.

pinterid committed:
Update exports

The achievement tab is now being exported.

pinterid committed:
Implement achievements tab

The achievements tab is now being implemented.

pinterid committed:
Update .env.template

Flags were added to the template.

pinterid committed:
Add path explorer

The path explorer flag was added.

pinterid committed:
Add inspector

The inspector flag was added.

pinterid committed:
Add console user

The console user flag was added.

Aichnerc committed:
Fix image modal

The image size has been adjusted to stop distortions.

Aichnerc committed:
Fix settings

Nico can thank me.

pinterid committed:
Add photo map

The photo map component was added.

pinterid committed:
Implement photo maps

Photo maps is now implemented.

schettn committed:
Apply setting state error fix for all settings

Now the settings are adjustable again.

Aichnerc committed:
Implement instagram image selector

It is now possible to add and remove images with the instagram selector.

pinterid committed:
Update photo map

The photo map data and display was updated.

pinterid committed:
Merge remote-tracking branch 'origin/migrate-intelv2-aa1' into migrate-intelv2-aa1

schettn committed:
Add display map setting

Now the map is only displayed if displayMap is enabled.

pinterid committed:
Add user check

it is no checked if the user is logged in and if its his profile.

Aichnerc committed:
Fix connect modal

The modal should now only show up at the own page and now close without errors.

schettn committed:
Fix talks page and add comment functionallity

Now the talks page works again. 
It is also possible to leave a comment.

schettn committed:
Merge remote-tracking branch 'origin/migrate-intelv2-aa1' into migrate-intelv2-aa1

pinterid committed:
Update action

The action now returns an achievement.

pinterid committed:
Update achievement tab

The achievement tab was updated.

pinterid committed:
Merge branch 'add-achievements-aa1' into migrate-intelv2-aa1

schettn committed:
Add select image conditional rendering

Now the select images button is only displayed if at least
one instagram profile is connected.

schettn committed:
Merge remote-tracking branch 'origin/migrate-intelv2-aa1' into migrate-intelv2-aa1

schettn committed:
Switch to production engine

Now intel and client no longer uses a local version of the snek
engine but the production one on engine.snek.at.

schettn committed:
Update dependencies

Some dependencies have been updated due to security
vulnerabilities and intel and client releases.

schettn commented on src/Routes.js:
```diff
@@ -35,13 +35,13 @@ class Routes extends React.Component {
-          component={(props) => <TalkPage {...props} />}
+          path="/temp"
+          component={(props) => <TempPage {...props} />}
         />
```
All "Temp" component should be removed.

schettn commented on src/components/atoms/charts/ContribRadar/index.jsx:
```diff
@@ -103,40 +103,29 @@ class ContribRadar extends React.Component {
     let totalIssues = 0;
     let totalRequests = 0;
     let totalCommits = 0;
     let totalSources = 1;
```
@Aichnerc why is `totalSources = 1`. Shouldn't it be equals to 0?

schettn commented on src/components/molecules/modals/FollowModal/index.jsx:
```diff
@@ -0,0 +1,238 @@
+class FollowModal extends React.Component {
+  state = {
+    loggedUser: null,
+  };
```
I think it would be better not to store the whole loggedUser into the state.
We should load only required datasets into the state.

schettn commented on src/components/molecules/modals/FollowModal/index.jsx:
```diff
@@ -0,0 +1,238 @@
+
+        loggedUser.person.follows = follows;
+
+        this.setState({ loggedUser });
```
This should be reworked.

- Use the new state defined above.
- Then remove the for loop. This workaround is no longer required. (`loggedUser.follows` is readOnly)

schettn commented on src/components/molecules/modals/FollowModal/index.jsx:
```diff
@@ -0,0 +1,238 @@
+        }
+      }
+
+      loggedUser.person.follows = follows;
```
This should be reworked:

- Use the new state defined above
- Rework the remove algorithm as follows:
```
   var members = [
        {name: 'Anna', class: 'one'}, 
        {name: 'Bob', class: 'two'},  
        {name: 'Chuck', class: 'two'}];

   var idx = members.findIndex(p => p.class=="two");
   var removed = members.splice(idx,1);
     
   console.log(removed);
   console.log(members);
```

schettn commented on src/components/molecules/modals/FollowModal/index.jsx:
```diff
@@ -0,0 +1,238 @@
+      }
+    }
+
+    return false;
```
Rework with `arr.find()`

schettn commented on src/components/molecules/modals/LikesModal/index.jsx:
```diff
@@ -0,0 +1,180 @@
+
+//#region > Components
+/** @class A modal that shows a list of people that liked */
+class LikesModal extends React.Component {
+  state = {
+    loggedUser: null,
+  };
+
+  componentDidMount = () => {
+    const { loggedUser } = this.props;
+
+    this.setState({
+      loggedUser: loggedUser,
+    });
+  };
```
Rework like FollowModal.

schettn commented on src/components/organisms/tabs/AchievementsTab/index.jsx:
```diff
@@ -0,0 +1,146 @@
+          if (res.achievement !== null) {
+            let achievements = [];
+
+            for (let count in this.state.achievements) {
+              achievements.push(this.state.achievements[count]);
+            }
```
Why?

schettn reviewed (changes requested):
Have a look

pinterid committed:
Improve code quality

The code quality was improved due to a request by Codacy.

pinterid committed:
Merge remote-tracking branch 'origin/migrate-intelv2-aa1' into migrate-intelv2-aa1

schettn committed:
Remove temp page

The temp page which implemented all newer
components is removed now because it is no longer
required.

schettn committed:
Remove unused code in ContribRadar

This code was never used therefore I removed it.

schettn committed:
Enhance FollowModal performence

The new implementation is more suitable for larger ones
records but does not fully work yet

schettn committed:
Improve person page refreshing

The page refreshing has been improved.

schettn committed:
Refresh infocard when modal closes

Now every time handleModalClose is called,
the infocard gets new data.

schettn committed:
Remove console logs

All uncommented console logs have been removed.

kleberbaum reviewed (approved):
Should be ok to merge.
